cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -819,7 +827,7 @@ private KubernetesObject createJob(V1ObjectMeta metadata,
         .withBackoffLimit(0)
         .withNewTemplate()
         .withMetadata(metadata)
-        .withSpec(createPodSpec(V1Job.class, runtimeConfigLocation, runtimeSpecs, "Never",
```

Comment:
The argument was unused in the inner methos and IntelliJ was warning me about it. It made sense because the pod spec should not be different if its to be deployed as a StatefulSet, Deployment or Job. I removed the parameter from the innermost functions and the argument became unused in callers. So I removed it from there too.

These are private methods so it should not break any API.